src/core/algorithms/ucc/ucc.h

```diff
@@ -1,13 +1,23 @@
 #pragma once
 
 #include "model/table/vertical.h"
+#include <memory>
```

Comment:
```suggestion
#include <memory>

#include "model/table/vertical.h"
```

STL, external libraries, project files is the header order here, see [the style guide](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes). I don't know why `clang-format` didn't catch it.